Info about ghpages added in docs

souvik-databricks opened the pull request:
## Enriched the docs with ghpages information
- **_Added the ghpages badge on the top with the hyperlink pointing to the [tempo.github.io](tempo.github.io) the docs page itself_**
<img width="909" alt="image" src="https://user-images.githubusercontent.com/98817232/159087513-dd2cb250-1264-4a9f-b416-69807d4d46ea.png">  <img width="148" alt="image" src="https://user-images.githubusercontent.com/98817232/159088054-2bf42e24-5ade-4e0c-b288-a285edfeb18e.png">


- **_Added the docs description of how it is going to be maintained_** 
<img width="855" alt="image" src="https://user-images.githubusercontent.com/98817232/159088832-878d9910-b2fb-4a64-a870-06b61e568c62.png">



- **_Added a hyperlink to the sphinx page which shows the Github workflow YAML for deploying docs on ghpages_** 
<img width="855" alt="image" src="https://user-images.githubusercontent.com/98817232/159089119-9cfcdd11-8873-4bba-aad6-070fdd37c10d.png">

rportilla-databricks commented on README.md:
```diff
@@ -12,6 +12,7 @@ The purpose of this project is to make time series manipulation with Spark simpl
 [![Downloads](https://pepy.tech/badge/dbl-tempo/month)](https://pepy.tech/project/dbl-tempo)
 [![PyPI version](https://badge.fury.io/py/dbl-tempo.svg)](https://badge.fury.io/py/dbl-tempo)
+[![docs](https://img.shields.io/badge/docs-ghpages-lemon)](https://tempo.github.io)
 
```
This is awesome - I think the hyperlink is great - in terms of the text shown. Instead of ghpages, I wonder if we can display a status of the docs (e.g. passing instead of `ghpages`. Here is an example from dbx - .. |docs| image:: https://readthedocs.org/projects/dbx/badge/?version=latest
    :target: https://dbx.readthedocs.io/en/latest/?badge=latest
    :alt: Documentation Status)

souvik-databricks replied:
@rportilla-databricks 
Yeup makes sense to have a _docs deployment status_ instead of the text `docs | ghpages`. For DBX its easy to do as its hosted on readthedocs. 

But I got the point Ricardo, let me check with @tnixon on Monday, If we could have **_a separate workflow only for the sphinx deployment on github pages_** then we could have that _**docs workflows status badge**_ like we have for the build status <img width="94" alt="image" src="https://user-images.githubusercontent.com/98817232/159140284-a5311c3e-5a5b-4aab-9c34-3d3a61694193.png">

souvik-databricks replied:
@rportilla-databricks 

I have made the changes now we have the `docs | passing` badge
<img width="873" alt="image" src="https://user-images.githubusercontent.com/98817232/159570918-ac2b6891-0f4d-4b9a-935d-3183eb70b30f.png">

With the badge hyperlinked to the correct docs page also: 
<img width="239" alt="image" src="https://user-images.githubusercontent.com/98817232/159571069-cf2967c9-212a-4277-8815-05c6fd2b9272.png">

rportilla-databricks commented on docs/index.rst:
```diff
@@ -44,6 +47,8 @@ Tempo is very easy to use:
     display(phone_accel_tsdf)
 
+.. _direct-git-install:
+
```
I'm not sure why this line appears - can we remove it? Here is what I'm seeing as raw text in the index.rest and I think we could re-word it. I would suggest something like - time series on Spark & Photon is the highly performant for historical analyses. We are simplifying all the common functions to make development more easy.

[:ref:`Much of the Spark ecosystem already uses <who-uses>`](https://github.com/databrickslabs/tempo/blob/enrichDocHostingInfo/docs/index.rst#id1) tempo for time-series analysis and you should too.

souvik-databricks replied:
Yes of course we can rephrase the text content in the docs. I will make this change (rephrase the text with what you mentioned and I will also remove that reference that you see there `:ref:`). 

In fact once the docs workflow is setup by Tristan and we have it live. It would be helpful to have the team's feedback on all of the pages.

souvik-databricks replied:
The suggested change has been pushed.

rportilla-databricks commented on docs/index.rst:
```diff
@@ -79,9 +84,23 @@ pre-processing to time-series analytics & machine learning!
+Docs on Github Pages
+--------------------
+
+Tempo's `sphinx documentation is hosted on github pages <https://www.sphinx-doc.org/en/master/tutorial/deploying.html#id5>`_
```
Should we link to the actual tempo docs instead of generic sphinx docs? Or perhaps both would be useful actually.

souvik-databricks replied:
@rportilla-databricks Do you want me to setup one separate page which talks about how we are deploying our docs with sphinx using github-pages and then have the link point to that page? 
The only reason I didn't have too much details is because I thought that it's beyond the scope of our work in tempo and if anyone wants to build there docs using sphinx they can follow the sphinx generic docs.

rportilla-databricks replied:
No, this is fine.

souvik-databricks commented:
@rportilla-databricks 

Three more additional changes added to this PR:
* I have also fixed this environment variable issue which was causing the display method to not work in DCS clusters. Now we are using `DB_HOME` which is the unique environment key which will be present in both _DCS and Non - DCS Clusters._

* Added a note to the API docs in sphinx which lets users know that the standardization of the docstrings is underway and thanks them for the patience and support. <img width="914" alt="image" src="https://user-images.githubusercontent.com/98817232/159572057-a8946d03-0874-4384-b959-98fcbeefe703.png">

* Also fixed this variable dropdown which was not readable in dark theme in interpolation earlier <img width="914" alt="image" src="https://user-images.githubusercontent.com/98817232/159572305-ccf7ca07-0ab7-4581-af8a-662e0d1f278f.png"> and now we have it fixed <img width="914" alt="image" src="https://user-images.githubusercontent.com/98817232/159572404-9f9f1ffa-fad9-4aae-be2d-b3daecea71cc.png">

rportilla-databricks commented on python/tempo/utils.py:
```diff
@@ -8,7 +8,7 @@
 from pyspark.sql.dataframe import DataFrame
 
 logger = logging.getLogger(__name__)
```
Can we add a quick comment about this environment variable? What are those expected in the databricks environment? (just the key name)

souvik-databricks replied:
This comment has been added. After the interpolation docstring. The PR will be good to go ahead.
